app/src/main/java/org/breezyweather/BreezyWeather.kt

```diff
@@ -118,14 +122,29 @@ class BreezyWeather : Application(),
     }
 
     private fun setDayNightMode() {
-        AppCompatDelegate.setDefaultNightMode(
-            ThemeManager.getInstance(this).uiMode.value!!
-        )
+        updateDayNightMode(ThemeManager.getInstance(this).uiMode.value!!)
+        
         ThemeManager.getInstance(this).uiMode.observeForever {
-            AppCompatDelegate.setDefaultNightMode(it)
+            updateDayNightMode(it)
         }
     }
 
+    private fun updateDayNightMode(dayNightMode: Int) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            // Sets and persists the night mode setting for this app. This allows the system to know
+            // if the app wants to be displayed in dark mode before it launches so that the splash
+            // screen can be displayed accordingly.
+            ContextCompat.getSystemService(getApplication(this), UiModeManager::class.java)
+                ?.setApplicationNightMode(
```

Comment:
This fixes the follwing issues:
- On newer Android version a theme change would not immediately be applied in the location list (management fragment).
- With targetSDK/compileSDK 35 a theme change would cause the system bars to have a solid color.

Additionally, this is a prerequisite for #137.